### ISSUE TITLE
load_account UserWarning should have stacklevel=2

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-ibmq-provider#372]
When `load_account` is ran twice, a warning is triggered:
<img width="1014" alt="Screen Shot 2019-10-04 at 4 52 33 PM" src="https://user-images.githubusercontent.com/766693/66239694-90e0a580-e6c8-11e9-887a-dc43620d2271.png">

The last line in the message, is a bit confusing because looks like an incomplete duplication. I suggest to use `warnings.warn('Credentials are already in use. The existing account in the session will be replaced.', stacklevel=2)`, so the error would look like this:
<img width="1010" alt="Screen Shot 2019-10-04 at 4 58 23 PM" src="https://user-images.githubusercontent.com/766693/66239787-c71e2500-e6c8-11e9-9063-e0114b321f35.png">

**Comment by diego-plan9** [comment]
Thanks @1ucian0 ! Actually we should do some more thorough checking of other `warnings.warn` as well.

**Comment by luisg5** [comment]
Hello @1ucian0! In case you were wondering, the reason for the incomplete message was actually due to a weird formatting feature on part of the `warnings` module. 

I believe the `stacklevel` parameter only coincidentally removed the last part of the message. It's actual [use case](https://docs.python.org/3.5/library/warnings.html#warnings.warn) is to change the `caller` the warning refers to. If you look closely at your messages above, the first warning message states the warning was issued from `.../qiskit/providers/ibmq/ibmqfactory.py:181`. And, the second message states the warning was issued from `.../ipykernel_launcher.py:2`, which happens to be the caller of `IBMQ.load_account()` in your case.

**Comment by mtreinish** [comment]
Why are we using the warnings module here at all? This should just be a log message we can use the warn level which will still print to stderr in default logging config. Normally you only want to use a warning if the code being called has an issue (like a deprecation) which is why the caller is in the default format (and is an attribute of a warning message). Just accidentally loading credentials twice doesn't fit into this category. It's still runtime info you want to get to the user, but the normal mechanism for that I would use is logging. If you just replace:

https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/ibmqfactory.py#L150-L151

with:

```
logger.warn('Credentials are already in use. The existing '
            'account in the session will be replaced.')
````

it will work as expected still print the message to stderr, but not include any of the extra information about the caller (unless you add that to your log format)

The fix proposed in #379 is not a good approach we should not be changing the global log message formatting, especially in a library that has nothing to do with warnings. If you do not want to switch to logging `stacklevel=2` as @1ucian0 suggested is the correct fix here because it will point to the line which called load_account() by accident a second time. You don't want to say the source of the warning is the line internal to the provider because it's not loading the account twice.

**Comment by ajavadia** [comment]
Yeah can we please reconsider the use of warning here? It gets a bit annoying in interactive python sessions when you accidentally load account twice. What action does this warning actually prompt on the part of the user? I think a log is a more user-friendly way of doing this as @mtreinish suggested.

**Comment by diego-plan9** [comment]
@ajavadia the use of `warning` has been essentially carried over from the times where `load_account` was introduced - at that point in time, it was deemed preferable to use a `warning` instead of `logger.warning` on the grounds that it would be "more visible to users"; and it sort of stuck as the usual philosophy for the provider for similar messages and context.

Personally I also feel more comfortable with a plain `logger.warning` and placing a bit more trust on the users being slightly more familiar with the logging facilities - does your comment count as having your green light for adopting that approach? :crossed_fingers:

**Comment by ajavadia** [comment]
Yes that would be great thanks!
